### PR TITLE
SSCS-10076 Add get mapping for party notifications

### PR DESCRIPTION
**Pull request by MohammedHMCTS**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10076

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by nickwest1 — DISMISSED** [review]

**Review by wmarshid — APPROVED** [review]

**Review by wmarshid — APPROVED** [review]

**Review by nickwest1 — APPROVED** [review]

**Review by nickwest1 — APPROVED** [review]

**Review by suetlaiHMCTS — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingPartiesNotifiedApi.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    String SERVICE_AUTHORIZATION = "ServiceAuthorization";
+    String PARTIES_NOTIFIED_ENDPOINT = "/partiesNotified";
+    String ID = "id";
+
+    @GetMapping(value = PARTIES_NOTIFIED_ENDPOINT, consumes = MediaType.APPLICATION_JSON_VALUE)
+    PartiesNotified getPartiesNotifiedRequest(
+        @RequestHeader(AUTHORIZATION) String authorisation,
+        @RequestHeader(SERVICE_AUTHORIZATION) String serviceAuthorization,
+        @RequestParam(ID) String id,
```
Comment (gokul-sol) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingPartiesNotifiedApi.java:26]: @MohammedHMCTS Isn't this called hearingID in HMC?

Comment (wmarshid) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingPartiesNotifiedApi.java]: no that's in the response not for the request

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingPartiesNotifiedApi.java [hunk 2]
```diff
@@ -17,14 +15,13 @@ public interface HmcHearingPartiesNotifiedApi {
 
     String SERVICE_AUTHORIZATION = "ServiceAuthorization";
     String PARTIES_NOTIFIED_ENDPOINT = "/partiesNotified";
-    String ID = "id";
+    String ID = "hearingID";
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingPartiesNotifiedApi.java:18]: this value is wrong, should be "id" according to the swagger spec

https://hmcts.github.io/reform-api-docs/swagger.html?url=https://hmcts.github.io/reform-api-docs/specs/hmc-cft-hearing-service.json#/parties-notified-controller/getPartiesNotifiedUsingGET

File: src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/PartiesNotified.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.List;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class PartiesNotified {
+
+    private Long partiesNotifiedRequestId;
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/PartiesNotified.java:18]: We should change this variable name to "hearingID" to match the response object in the swagger spec, see below.

```
{
  "hearingID": "string",
  "responses": [
    {
      "partiesNotified": "2022-05-12T20:42:09.623Z",
      "requestVersion": 0,
      "responseReceivedDateTime": "2022-05-12T20:42:09.623Z",
      "responseVersion": 0,
      "serviceData": {}
    }
  ]
}
```

File: src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/PartiesNotified.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+    private List<PartiesNotifiedResponse> partiesNotifiedResponse;
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/PartiesNotified.java:20]: We should change this variable name to "responses" to match the swagger spec.

File: src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/PartiesNotifiedResponse.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.joda.time.DateTime;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class PartiesNotifiedResponse {
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/PartiesNotifiedResponse.java:15]: Minor: can we change the ordering of these to match the swagger spec (usual json convention is alphabetical order as well like the spec has)

**Review by wmarshid — CHANGES_REQUESTED** [review]

**Review by wmarshid — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingPartiesNotifiedApi.java (hunk 1 above)
Comment (gokul-sol) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingPartiesNotifiedApi.java]: @wmarshid ah got you thanks.

File: src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/PartiesNotified.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.List;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class PartiesNotified {
+
+    private Long hearingID;
```
Comment (gokul-sol) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/PartiesNotified.java:18]: @MohammedHMCTS this variable type for hearingID is **String**
HMC PR fyi https://github.com/hmcts/hmc-cft-hearing-service/pull/193/files#diff-a6cb9f13bf2b4ae9b10fae6e8c5db14121cf1905b6bdfd1092d3d0b5dcfb4e81R12

Comment (MohammedHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/PartiesNotified.java]: Good spot I updated that, I was aware of the change to String but forgot to update it... Sort now :)

**Review by nickwest1 — APPROVED** [review]

**Review by wmarshid — APPROVED** [review]

**Pull request merged**